Dependencies.cmake

```diff
@@ -33,7 +33,9 @@ function(cloysterhpc_setup_dependencies)
 
   if(NOT TARGET spdlog::spdlog)
     if (cloysterhpc_ENABLE_CONAN)
-      CPMFindPackage(NAME spdlog)
+      CPMFindPackage(NAME spdlog
+       OPTIONS
+        "SPDLOG_FMT_EXTERNAL ON")
```

Comment:
Does that works when using Conan instead of CPM?

Reply:
Probably not... but I used conan and it worked.

Reply:
Well whatever. That may be an issue at some point but may be not. If not we gonna open a bug for this.

The root of the problem here is that CPM seems to be the way forward for Package Management, but there's some packages that aren't available yet. So we rely on Conan for it.

A bigger issue will be when we move to `std::format`... what if this PR breaks with it?

Reply:
I will look if CPM supports adding extra parameters

Regarding `std::format`, it supports formatting chrono data types ([source](https://en.cppreference.com/w/cpp/utility/format/format)):

> For chrono types, the format specification is interpreted as [chrono format specification](https://en.cppreference.com/w/cpp/chrono/system_clock/formatter#Format_specification).